### PR TITLE
improvement: add category for some functions

### LINKED ISSUE #4615: Add category for some functions
**Summary**

Empty category functions:
```
mysql> select name,category from system.functions where category='';
+-------------------------+----------+
| name                    | category |
+-------------------------+----------+
| tostartoftenminutes     |          |
| todayofmonth            |          |
| try_inet_aton           |          |
| inet_ntoa               |          |
| tuple                   |          |
| ln                      |          |
| touint64                |          |
| log10                   |          |
| rtrim                   |          |
| try_parse_json          |          |
| octet_length            |          |
| unhex                   |          |
| tostartofmonth          |          |
| toint8                  |          |
| ipv4numtostring         |          |
| toint16                 |          |
| not_in                  |          |
| touint16                |          |
| ascii                   |          |
| or                      |          |
| tofloat32               |          |
| tohour                  |          |
| bin                     |          |
| toint64                 |          |
| format                  |          |
| timeslot                |          |
| and                     |          |
| in                      |          |
| tomonday                |          |
| trim                    |          |
| try_inet_ntoa           |          |
| siphash64               |          |
| todate                  |          |
| log2                    |          |
| parse_json              |          |
| toyear                  |          |
| xor                     |          |
| tominute                |          |
| todate16                |          |
| touint8                 |          |
| tonull                  |          |
| exists                  |          |
| ltrim                   |          |
| touint32                |          |
| power                   |          |
| tostartofquarter        |          |
| not                     |          |
| todatetime64            |          |
| log                     |          |
| tofloat64               |          |
| tostartofsecond         |          |
| tostring                |          |
| ignore                  |          |
| hex                     |          |
| inet_aton               |          |
| oct                     |          |
| ipv4stringtonum         |          |
| abs                     |          |
| repeat                  |          |
| todayofweek             |          |
| sha                     |          |
| tryipv4stringtonum      |          |
| tryipv4numtostring      |          |
| tostartofhour           |          |
| toboolean               |          |
| todatetime32            |          |
| tostartofday            |          |
| todatetime              |          |
| tostartoffiveminutes    |          |
| tostartoffifteenminutes |          |
| todate32                |          |
| today                   |          |
| current_user            |          |
| tostartofweek           |          |
| toint32                 |          |
| todayofyear             |          |
| tosecond                |          |
| tostartofminute         |          |
| city64withseed          |          |
| tomonth                 |          |
| quote                   |          |
| multiply                |          |
| addyears                |          |
| -                       |          |
| !=                      |          |
| addhours                |          |
| /                       |          |
| regexp                  |          |
| adddays                 |          |
| %                       |          |
| subtractmonths          |          |
| >                       |          |
| like                    |          |
| subtractyears           |          |
| negate                  |          |
| >=                      |          |
| minus                   |          |
| addmonths               |          |
| not rlike               |          |
| addseconds              |          |
| <=                      |          |
| not like                |          |
| modulo                  |          |
| +                       |          |
| addminutes              |          |
| subtractminutes         |          |
| subtractdays            |          |
| <>                      |          |
| rlike                   |          |
| plus                    |          |
| divide                  |          |
| div                     |          |
| *                       |          |
| not regexp              |          |
| subtractseconds         |          |
| <                       |          |
| =                       |          |
| subtracthours           |          |
| mod                     |          |
| uniq                    |          |
| std                     |          |
| covar_pop               |          |
| covar_samp              |          |
| stddev                  |          |
+-------------------------+----------+
124 rows in set (0.02 sec)
Read 218 rows, 59.47 KB in 0.013 sec., 16.31 thousand rows/sec., 4.45 MB/sec.

```

Category functions:
```
mysql> select name,category from system.functions where category!='' order by category;
+-------------------+-------------+
| name              | category    |
+-------------------+-------------+
| avg               | aggregate   |
| argmin            | aggregate   |
| windowfunnel      | aggregate   |
| stddev_pop        | aggregate   |
| count             | aggregate   |
| sum               | aggregate   |
| max               | aggregate   |
| min               | aggregate   |
| argmax            | aggregate   |
| isnotnull         | conditional |
| if                | conditional |
| isnull            | conditional |
| database          | context     |
| version           | context     |
| toyyyymm          | datetime    |
| yesterday         | datetime    |
| tostartofisoyear  | datetime    |
| tostartofyear     | datetime    |
| tomorrow          | datetime    |
| toyyyymmdd        | datetime    |
| toyyyymmddhhmmss  | datetime    |
| now               | datetime    |
| md5               | hash        |
| xxhash64          | hash        |
| xxhash32          | hash        |
| siphash           | hash        |
| sha1              | hash        |
| sha2              | hash        |
| blake3            | hash        |
| atan              | numeric     |
| ceiling           | numeric     |
| truncate          | numeric     |
| cos               | numeric     |
| sqrt              | numeric     |
| radians           | numeric     |
| pow               | numeric     |
| crc32             | numeric     |
| asin              | numeric     |
| exp               | numeric     |
| floor             | numeric     |
| tan               | numeric     |
| sign              | numeric     |
| degrees           | numeric     |
| rand              | numeric     |
| pi                | numeric     |
| sin               | numeric     |
| atan2             | numeric     |
| cot               | numeric     |
| ceil              | numeric     |
| round             | numeric     |
| acos              | numeric     |
| totypename        | other       |
| runningdifference | other       |
| instr             | string      |
| reverse           | string      |
| mid               | string      |
| ucase             | string      |
| to_base64         | string      |
| substring_index   | string      |
| char_length       | string      |
| length            | string      |
| field             | string      |
| substring         | string      |
| soundex           | string      |
| left              | string      |
| char              | string      |
| character_length  | string      |
| strcmp            | string      |
| lcase             | string      |
| rpad              | string      |
| upper             | string      |
| concat            | string      |
| bit_length        | string      |
| ord               | string      |
| space             | string      |
| replace           | string      |
| export_set        | string      |
| find_in_set       | string      |
| substr            | string      |
| lpad              | string      |
| elt               | string      |
| locate            | string      |
| from_base64       | string      |
| position          | string      |
| concat_ws         | string      |
| regexp_like       | string      |
| lower             | string      |
| right             | string      |
| insert            | string      |
| sleep             | test        |
| generateuuidv4    | uuid        |
| isnotemptyuuid    | uuid        |
| zerouuid          | uuid        |
| isemptyuuid       | uuid        |
+-------------------+-------------+
94 rows in set (0.04 sec)

```

### PR DESCRIPTION
**Pull request by jiahui-97**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Changelog

- Improvement

## Related Issues

Part of  #4615

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]
Good

**Comment by jiahui-97** [comment]
There are still 103 functions without category, I will add them later.

**Comment by Xuanwo** [comment]
Ask @sundy-li for more information.

Should those function docs be auto-generated instead?

**Comment by sundy-li** [comment]
> Should those function docs be auto-generated instead?

Actually, we generate the `system.functions` from markdown documents.
We have investigated writing documents inside rust codes. But it may bring much more heavy work to rust compiler.

So markdown files are doc templates now.

See: https://github.com/datafuselabs/databend/blob/4abb5c3177f75ef3f0f031f574d0e7c2e987e522/common/functions/src/rdoc/function_doc_asset.rs

**Comment by jiahui-97** [comment]
I tried to generate documents automatically. Most of the functions can correspond to https://dev.mysql.com/doc/refman/8.0/en/built-in-function-reference.html, but only based on this, it is a bit difficult to achieve automatic generation logic such as syntax and examples.

So I prepare to generate documentation manually.

File: docs/doc/30-reference/20-functions/60-conversion-functions/tofloat32.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+title: toFloat32
```
Comment (sundy-li) [inline docs/doc/30-reference/20-functions/60-conversion-functions/tofloat32.md:2]: Hello, I think we should just keep `cast.md` doc and remove all `toxxx.md`

1. Use `title_includes` to cover all other aliases, eg: https://github.com/datafuselabs/databend/blob/f799b6e6114d00cb48d49f9c611105c052bf78de/docs/doc/30-reference/20-functions/30-datetime-functions/subtractinterval.md

2. Create a link to indicate the datatypes that we supported in `cast.md`:  https://github.com/datafuselabs/databend/blob/8dcb757f2da1658f12e4854cf0b14dee7099e8f8/docs/doc/30-reference/10-data-types/index.md

 

```

```

**Comment by jiahui-97** [comment]
There are still three functions without category. 

```sql
MySQL [(none)]> select name,category from system.functions where category='' order by  name;
+------+----------+
| name | category |
+------+----------+
| <    |          |
| <=   |          |
| <>   |          |
+------+----------+
```

The mistake happens when parsing "title_includes: =, >=, >, !=, <=, <, <>". if text has '<', pulldown_cmark can't put the entire line into txt. And I don't know how to fix it. 

Test case. 
```rust
use pulldown_cmark::{Event, Options, Parser};

fn main() {
    let markdown_input: &str = "title_includes: =, >=, >, !=, <=, <, <>";
    println!("Parsing the following markdown string:\n{}", markdown_input);

    let parser = Parser::new_ext(markdown_input, Options::empty());

    let mut inside_head = false;
    for event in parser {
        match event {
            Event::Text(txt) => {
                let txt = txt.as_ref();
                if inside_head && txt.starts_with("title_includes:") {
                    println!("original txt: {}", txt);
                } else {
                    println!("others: {}", txt)
                }
            }
            Event::Start(_) => {
                inside_head = true;
            },
            Event::End(_) => {},
            Event::Code(_) => todo!(),
            Event::Html(_) => todo!(),
            Event::FootnoteReference(_) => todo!(),
            Event::SoftBreak => todo!(),
            Event::HardBreak => todo!(),
            Event::Rule => todo!(),
            Event::TaskListMarker(_) => todo!(),
        }
    }
}

Parsing the following markdown string:
title_includes: =, >=, >, !=, <=, <, <>
original txt: title_includes: =, >=, >, !=, 
others: <
others: =, 
others: <
others: , 
others: <
others: >
```

**Comment by sundy-li** [comment]
> There are still three functions without category.
> 
> ```sql
> MySQL [(none)]> select name,category from system.functions where category='' order by  name;
> +------+----------+
> | name | category |
> +------+----------+
> | <    |          |
> | <=   |          |
> | <>   |          |
> +------+----------+
> ```
> 
> The mistake happens when parsing "title_includes: =, >=, >, !=, <=, <, <>". if text has '<', pulldown_cmark can't put the entire line into txt. And I don't know how to fix it.
> 
> Test case.
> 
> ```rust
> use pulldown_cmark::{Event, Options, Parser};
> 
> fn main() {
>     let markdown_input: &str = "title_includes: =, >=, >, !=, <=, <, <>";
>     println!("Parsing the following markdown string:\n{}", markdown_input);
> 
>     let parser = Parser::new_ext(markdown_input, Options::empty());
> 
>     let mut inside_head = false;
>     for event in parser {
>         match event {
>             Event::Text(txt) => {
>                 let txt = txt.as_ref();
>                 if inside_head && txt.starts_with("title_includes:") {
>                     println!("original txt: {}", txt);
>                 } else {
>                     println!("others: {}", txt)
>                 }
>             }
>             Event::Start(_) => {
>                 inside_head = true;
>             },
>             Event::End(_) => {},
>             Event::Code(_) => todo!(),
>             Event::Html(_) => todo!(),
>             Event::FootnoteReference(_) => todo!(),
>             Event::SoftBreak => todo!(),
>             Event::HardBreak => todo!(),
>             Event::Rule => todo!(),
>             Event::TaskListMarker(_) => todo!(),
>         }
>     }
> }
> 
> Parsing the following markdown string:
> title_includes: =, >=, >, !=, <=, <, <>
> original txt: title_includes: =, >=, >, !=, 
> others: <
> others: =, 
> others: <
> others: , 
> others: <
> others: >
> ```

Ok, I think we can leave these, they are trivial and commonly used.

**Comment by BohuTANG** [comment]
Some documents issue, we can check it：
cd website
yarn install
yarn build

File: docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md [hunk 2]
```diff
@@ -1,25 +1,143 @@
...
+title_includes: toBoolean, toDate, toDateTime, toTimestamp, toFloat32, toFloat64, toInt8, toInt16, toInt32, toInt64, toNull, toString, toUInt8, toUInt16, toUInt32, toUInt64
 ---
 
-to_int(8|16|32|64)
+Type conversion to target type. 
 
 ## Syntax
 
 ```sql
-to_int8(expr) — Results in the Int8 data type.
```
Comment (sundy-li) [inline docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md:10]: The latest style is `to_intxx` after we merged #5126

**Comment by BohuTANG** [comment]
Thank you @jiahui-97 for the contribution!

File: docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md (hunk 2 above)
Comment (jiahui-97) [reply docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md:10]: I noticed that, so I tried and found to_int8 can't work.

```sql 
MySQL [(none)]> select to_int8(1);
ERROR 1105 (HY000): Code: 2602, displayText = Unknown Function to_int8 (while in analyze select projection).
MySQL [(none)]> select toint8(1);
+-----------+
| toint8(1) |
+-----------+
|         1 |
+-----------+
```

Comment (jiahui-97) [reply docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md:10]: Sorry, maybe I don't get the latest version. I'll try it again.

Comment (BohuTANG) [reply docs/doc/30-reference/20-functions/60-conversion-functions/type-conversion.md:10]: I am working on it, and some other improvements will be opened a PR soon.

**Pull request merged**